dao/src/main/scala/za/co/absa/enceladus/dao/MenasException.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.dao
+
+abstract class MenasException(message: String, cause: Throwable) extends Exception(message, cause)
+
+abstract class RetryableException(message: String, cause: Throwable) extends MenasException(message, cause)
+
+abstract class OptionallyRetryableException(message: String, cause: Throwable) extends MenasException(message, cause)
+
+object RetryableException {
```

Comment:
Okay, I have no objections to that. 

I usually preferred to implement abstract classes and interfaces on the top, and then specific implementations on the bottom of a file, so that a collection of all abstract things can be seen easily. But I guess that there are no good arguments pro or against this so I'm happy to follow whichever style you/team prefer.